internal/template/template.go

```diff
@@ -144,6 +149,25 @@ func (e *Engine) RenderMetricQueries(metric *redskyv1beta1.Metric, trial *redsky
 	return b1.String(), b2.String(), nil
 }
 
+const rsoProm = "http://{{ promServer .ExperimentNamespacedName }}:9090"
+
+// RenderPrometheusURL returns the URL for the RSO managed prometheus instance
+func (e *Engine) RenderPrometheusURL(metric *redskyv1beta1.Metric, trial *redskyv1beta1.Trial) (string, error) {
+	data := newMetricData(trial, nil)
+
+	urlBytes, err := e.render("url", rsoProm, data)
+
+	return urlBytes.String(), err
+}
+
+func (e *Engine) RenderAnnotationTarget(metric *redskyv1beta1.Metric, trial *redskyv1beta1.Trial) (string, error) {
+	data := newMetricData(trial, nil)
+
+	labelPairBytes, err := e.render("annotation", metric.Query, data)
+
+	return labelPairBytes.String(), err
+}
+
```

Comment:
How are these getting used? I don't think it necessary to leverage templates for this.

With the existing Prometheus metric collection, just change it so if there is nothing specified (i.e. today no labels, in the future, no URL) we just default the Prometheus URL to what we would be standing up. That is, instead of failing when we see missing configuration we will push the potential failure further down the road (when we try to connect).